frontend/lib/src/WidgetStateManager.ts

```diff
@@ -670,6 +673,30 @@ export class WidgetStateManager {
     })
   }
 
+  /**
+   * A dialog should only open if it is able to acquire the lock first.
+   * This ensures that only a single dialog is open any given time.
+   * The dialog must call {@link releaseDialogLock} when it closes, to allow
+   * other dialogs from opening.
+   *
+   * @returns true if the lock was acquired, false otherwise
+   */
+  public tryAcquireDialogLock(): boolean {
```

Comment:
Curious, this won't apply to our existing dialogs. Are we confirming that the Streamlit standard dialogs are allowed to appear above. Just want to make sure we  think about this correctly.

Reply:
In general the point of alignment is a great one! In this case I think this has been less of a focus for the other Dialogs as they are not triggerable by the user app code itself (unless they are and I don't know it 😅). I think we should use a similar logic (does not have to be this one 🙂) for all user-triggerable dialogs or whatever, but should descope the PR to exclude the existing "system" dialogs as much as possible and rather should do a separate follow-up project in order to keep the PR scope contend (I know there were some updates already especially with styling etc.)